### ISSUE TITLE
Extension support

### ISSUE THREAD

**Issue by PalmerAL** [PalmerAL/browser#1]
- [ ] create an extension store
 - [ ] create a way to manage extensions
 - [ ] create extension apis

**Comment by samuelmaddock** [comment]
Hey, @PalmerAL. Just wanted to let you know I've been working on improving support for Chrome extensions in Electron officially, current wip at electron/electron#17440. Please get in touch if you'd like to help out, my email is in my profile.

**Comment by AwesomeTech10** [comment]
Check out [Wexond](https://github.com/wexond/wexond), it has _partial_ support for Chrome extensions.

**Comment by dlobue** [comment]
FYI looks like there is a library for this now: https://www.npmjs.com/package/electron-extensions

**Comment by PalmerAL** [comment]
Yeah, that does look like it could potentially be a good option, although I haven't looked into it much yet.

**Comment by phenax** [comment]
Any updates on this?

**Comment by PalmerAL** [comment]
Not really. I'm not sure exactly what's happening with Electron's built-in support for this at the moment, but I don't think it's fully complete yet.

**Comment by AwesomeTech10** [comment]
https://www.electronjs.org/docs/api/session#sesloadextensionpath

**Comment by ndrogers** [comment]
Status?

**Comment by PalmerAL** [comment]
Extension support in Electron is still incomplete as far as I know.

The electron-extensions library won't work because of this: https://github.com/sentialx/electron-extensions/blob/master/PATENTS

So it's unlikely to happen soon unless someone wants to write another extension library - if that's something you'd be interested in doing, let me know!

Expanding the capabilities of userscripts is probably a more viable path forward.

**Comment by AdityaCyberSafe** [comment]
Hi PalmerAL,
I just wanted to say that I found an library that can add partial support for extensions.
Here's the link: [](https://github.com/sentialx/electron-extensions)

**Comment by AdityaCyberSafe** [comment]
Hi PalmerAL,
I just wanted to say that I found an library that can add partial support for extensions.
Here's the [link](https://github.com/sentialx/electron-extensions)

**Comment by shalva97** [comment]
are there any updates on this?

**Comment by PalmerAL** [comment]
Not really - still the same problems as before. Adding extension support (and then maintaining it and fixing compatibility issues with extensions) is a massive amount of work; Electron has some extension support, but it's incomplete. Extending the capabilities of userscripts is going to be easier, but means that existing Chrome extensions can't be re-used.

**Comment by fP4rad0x** [comment]
If it got extensions it would be amazing!

**Comment by 1Spinne** [comment]
pleeeease add the support for schrome extensions
they have th best extensions and i xant just use userscripts

**Comment by Zaki-Dz** [comment]
Please add chrome extension's support and the ability to import bookmarks from other browsers after that it will be the best browser i've ever used .

**Comment by alminisl** [comment]
I would love to have at least Dark Reader integration or something like that.

**Comment by ForumPlayer** [comment]
> I would love to have at least Dark Reader integration or something like that.

Yeah, I have a light sensitivity (aka Photophobia) and the Dark Reader extension is basic thing for me.
I have some strong headaches when i'm exposed to bright lights for more than a minute, so that's also why i use to work with dimmed lights

**Comment by PalmerAL** [comment]
For dark reader specifically, there's a userscript you can install that does something similar: https://github.com/PalmerAL/min-dark-mode-userscript

**Comment by hathtat** [comment]
Metamask integration built in?

**Comment by rewkha** [comment]
usercripts-like extensions? I mean like simple way to add them. And something like list on the min webpage?

**Comment by ForumPlayer** [comment]
@rewkha i don't think it won't work well, since the userscripts are kinda limited.
it's like having a native "Tampermonkey"

**Comment by Sanix-Darker** [comment]
This browser is amazing and i think, this is the only thing left for it to be the GOAT !
As far as i can read... it fully depends on electron support for extensions ?
I would like to contribute if it's possible !

**Comment by raphael10-collab** [comment]
Any news for supporting Chrome Extensions?

**Comment by re-kaptcha** [comment]
The only thing holding me back from setting Min as my default browser is being able to install chrome extensions. Electron already has support for extensions (see [here](https://www.electronjs.org/docs/latest/api/extensions)), although it's limited to unpacked extensions, not standard `.crx` Chrome extension files. However, since `crx` files are essentially the same as zip files, any standard tool, such as 7zip, can be used to unpack the extensions. 

The challenge would be to keep a repository of unzipped Chrome extensions for use, but, since this browser seems to be developer oriented in my opinion, it can be a simple git repo, or something along the lines of how `nix` or `homebrew` keep track of their packages.

A repository by the name of `min-extensions` or something could be created which hosts instructions on how to install specific packages. Then, a separate CLI tool, maybe included in Min, could would be responsible for instructions from the repo, and installing an extension onto the local system, which could then be used inside the Min browser.

**Comment by maisonhai3** [comment]
Without extension support, I find no motivation to use this browser.
Since my work is now heavily dependent on those extensions: Vimium, Grammarly, Workona, SendToKindle, and SendToNotion..

**Comment by TheBlckbird** [comment]
@kaptcha0 you could do the unzipping programmatically

**Comment by kaptcha0** [comment]
@TheBlckbird True, but the reason I proposed a git repo is so Min users can only install extensions that Min supports. I suppose one could do the unzipping at runtime, but some extensions might conflict with the browser.

**Comment by KSaiAkshit** [comment]
Any updates on this? Min is otherwise perfect!

**Comment by PixelatedLagg** [comment]
It's been 9 years. Can there be a decision if extensions are gonna be added or not?

**Comment by ioc2e3** [comment]
> Any updates on this? Min is otherwise perfect!

Agree ...

( block  ads ) and (prevent fingerprint track ) ...

These two types of extensions ...

.

**Comment by ioc2e3** [comment]
> Any updates on this? Min is otherwise perfect!

Agree ...

.........................

And also ...

( block  ads ) and (prevent fingerprint track ) ...

Hope to provide ...

These two types of extensions ...

.
